src/serac/infrastructure/output.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/infrastructure/output.hpp"
+
+#include "serac/infrastructure/initialize.hpp"
+#include "serac/infrastructure/logger.hpp"
+#include "serac/infrastructure/terminator.hpp"
+#include "serac/physics/utilities/finite_element_state.hpp"
+
+#include "conduit/conduit.hpp"
+#include "ascent/ascent.hpp"
+#include "axom/sidre.hpp"
+
+#include "mpi.h"
+
+namespace serac::output {
+
+void outputFields(const axom::sidre::DataStore& datastore, const std::string& file_name_prefix, double time,
+                  const Language language)
+{
+  SLIC_INFO(fmt::format("Outputting field data at time: {}", time));
+
+  // Configure Ascent to extract data
+  ascent::Ascent ascent;
+  conduit::Node  ascent_opts;
+  // Use MPI function that always returns an int type for the communicator
+  ascent_opts["mpi_comm"] = MPI_Comm_c2f(MPI_COMM_WORLD);
+  // Do not allow Ascent to use a local file to override actions
+  ascent_opts["actions_file"] = "";
+  ascent.open(ascent_opts);
+
+  std::string output_language = "";
+  if (language == Language::JSON) {
+    output_language = "json";
+  } else if (language == Language::HDF5) {
+    output_language = "hdf5";
+  } else if (language == Language::YAML) {
+    output_language = "yaml";
+  }
+
+  conduit::Node extracts;
+  extracts["e1/type"]            = "relay";  // relay is the saving extract
```

Comment:
? I'm not sure what either `relay` or `saving extract` means without looking at the ascent docs.